### PR TITLE
progress on manifest resolution

### PR DESCRIPTION
**Pull request by aakropotkin**
This completes a large chunk of manifest -> lockfile resolution.

I'd like to merge this chunk before handling _multiple systems_, _groups_, or existing lockfiles so that some of these structures can unblock other PRs and avoid complex merge conflicts and/or redundant work.

### CODE REVIEW

File: include/flox/resolver/manifest.hh [hunk 1]
```diff
@@ -119,6 +121,178 @@ from_json( const nlohmann::json & jfrom, ManifestRaw & manifest );
 // void to_json( nlohmann::json & jto, const ManifestRaw & manifest );
 
 
+/* -------------------------------------------------------------------------- */
+
+class UnlockedManifest
```
Comment (zmitchell) [inline include/flox/resolver/manifest.hh:126]: Can you add a docstring somewhere that explains what the difference is between a Manifest, a ManifestRaw, and an UnlockedManifest?

Comment (aakropotkin) [reply include/flox/resolver/manifest.hh:126]: https://imgur.com/gallery/NkAY5KS

Comment (aakropotkin) [reply include/flox/resolver/manifest.hh:126]: Yeah tho I will during regular business hours

Comment (aakropotkin) [reply include/flox/resolver/manifest.hh:126]: You're just witnessing iteration in action. I'll end up killing 2/3 of these structures later

Comment (zmitchell) [reply include/flox/resolver/manifest.hh:126]: Alright, I'll approve for now, just remember there's other people that will have to decipher once you're on PTO so leave us some breadcrumbs

**Review by zmitchell — APPROVED** [review]

File: include/flox/resolver/manifest.hh (hunk 1 above)
Comment (aakropotkin) [reply include/flox/resolver/manifest.hh:126]: Oh I know, and I'll do everything my faculties will allow to help them 🖤

**Pull request merged**